Code_Saeyeon_review_par_Cedric.py

```diff
@@ -0,0 +1,149 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Aug  1 16:29:27 2022
+
+@author: Saeyeon
+"""
+
+# %% Plot fig 5C (within-subject correlations with observed data, shuffing keeping pairs, full shuffling)
+
+
+#add a column in subj_correlations dataframe corresponding to the index (subject names)
+subj_correlations['subject name'] = list_subject
+
+#create a new dataframe to get accuracies for each subject, at each trial ("the accuracy of probability estimates was computed at the trial level as the distance between the subject's and IO's estimates)   
+subj_accuracies  = pd.DataFrame(index = data['subject'], 
+                                columns= ['subject name', 'accuracy_estimation', 'accuracy_confidence','accuracy_estimation_pairs_shuffling','accuracy_confidence_pairs_shuffling', 
+                                          'accuracy_estimation_full_shuffling','accuracy_confidence_full_shuffling'])
+subj_accuracies['subject name'] = data['subject'].values
+
+for i in range(len(data)):
```

Comment:
Instead of iterating over rows in this way, I think it would be possible to work with numpy arrays and use array operations to compute everything at once (cleaner and more efficient). E.g.:
```subj_accuracies['accuracy_estimation'] = np.abs(data['io_pred'].to_numpy() - data['sub_pred'].to_numpy())```